packages/babel-plugin/src/utils/normalize-props-usage.ts

```diff
@@ -15,18 +15,129 @@ const buildObjectChain = (path: NodePath<t.Node> | null, currentChain: string[]
     const { parentPath } = path;
     const { node } = parentPath;
 
+    // Skip over any default parameters when traversing,
+    // otherwise they mess up the recursion
+    if (t.isAssignmentPattern(node)) {
+      return buildObjectChain(parentPath, currentChain);
+    }
+
+    // e.g. an arrow function like
+    // ({ a: [, second] }) => ...
+    if (t.isArrayPattern(node)) {
+      throw new Error(
+        'Compiled does not support arrays given in the parameters of an arrow function.'
+      );
+    }
+
     if (t.isObjectProperty(node) && t.isIdentifier(node.key)) {
       currentChain.unshift(node.key.name);
     }
 
-    return buildObjectChain(parentPath.parentPath, currentChain);
+    // When the listKey of the current path is 'params', this means
+    // that the current path === the entire props object (e.g. { a, b = 5 }).
+    // So we stop recursively traversing after we pass this point.
+    //
+    // (Note that we are recursing upwards two parents at a time.)
+    if (parentPath.listKey !== 'params') {
+      return buildObjectChain(parentPath.parentPath, currentChain);
+    }
   }
 
   currentChain.unshift(PROPS_IDENTIFIER_NAME);
 
   return currentChain;
 };
 
+/**
+ * Given an object pattern (e.g. { width, height = 16 }) from the
+ * parameters of an arrow function, return an object with the
+ * default parameters (if any) found. For example, { height: 16 }.
+ *
+ * @param path a deconstructed object within the function parameters
+ * @returns an object where the keys are arguments with default
+ *          parameters, and the values are their default values
+ */
+const getDefaultParameters = (
+  path: NodePath<t.Identifier | t.RestElement | t.Pattern>
+): Record<string, t.Expression> => {
+  const node = path.node;
+  const assignments: Record<string, t.Expression> = {};
+
+  const FindAllAssignmentsVisitor = {
+    AssignmentPattern(path: NodePath<t.AssignmentPattern>) {
+      const { node } = path;
+      if (t.isIdentifier(node.left)) {
+        assignments[node.left.name] = node.right;
+      } else {
```

Comment:
I couldn't think of any valid examples that would cause these two errors (lines 72 and 85) to be thrown 🤔 However, it's nonetheless theoretically possible according to the babel types